pkg/connector/connector.go

```diff
@@ -102,45 +114,73 @@ func (c *Okta) Metadata(ctx context.Context) (*v2.ConnectorMetadata, error) {
 }
 
 func (c *Okta) Validate(ctx context.Context) (annotations.Annotations, error) {
-	l := ctxzap.Extract(ctx)
-	token := newPaginationToken(defaultLimit, "")
-
-	_, respCtx, err := getOrgSettings(ctx, c.client, token)
-	if err != nil {
-		return nil, fmt.Errorf("okta-connector: verify failed to fetch org: %w", err)
+	if c.apiToken != "" {
+		l := ctxzap.Extract(ctx)
+		token := newPaginationToken(defaultLimit, "")
+
+		_, respCtx, err := getOrgSettings(ctx, c.client, token)
+		if err != nil {
+			return nil, fmt.Errorf("okta-connector: verify failed to fetch org: %w", err)
+		}
+
+		_, _, err = parseResp(respCtx.OktaResponse)
+		if err != nil {
+			return nil, fmt.Errorf("okta-connector: verify failed to parse response: %w", err)
+		}
+
+		if respCtx.OktaResponse.StatusCode != http.StatusOK {
+			err := fmt.Errorf("okta-connector: verify returned non-200: '%d'", respCtx.OktaResponse.StatusCode)
+			l.Error("Invalid Status Code from Verity", zap.Error(err))
+			return nil, err
+		}
 	}
 
-	_, _, err = parseResp(respCtx.OktaResponse)
-	if err != nil {
-		return nil, fmt.Errorf("okta-connector: verify failed to parse response: %w", err)
-	}
-
-	if respCtx.OktaResponse.StatusCode != http.StatusOK {
-		err := fmt.Errorf("okta-connector: verify returned non-200: '%d'", respCtx.OktaResponse.StatusCode)
-		l.Error("Invalid Status Code from Verity", zap.Error(err))
-		return nil, err
-	}
 	return nil, nil
 }
 
 func (c *Okta) Asset(ctx context.Context, asset *v2.AssetRef) (string, io.ReadCloser, error) {
 	return "", nil, fmt.Errorf("not implemented")
 }
 
-func New(ctx context.Context, domain, apiToken string, syncInactiveApps bool) (*Okta, error) {
+func New(ctx context.Context, domain, apiToken,
+	oktaClientId, oktaPrivateKey, oktaPrivateKeyId string,
+	syncInactiveApps, provisioningEnabled bool) (*Okta, error) {
+	var (
+		oktaClient *okta.Client
+		scopes     = defaultScopes
+	)
 	client, err := uhttp.NewClient(ctx, uhttp.WithLogger(true, nil))
 	if err != nil {
 		return nil, err
 	}
 
-	_, oktaClient, err := okta.NewClient(ctx,
-		okta.WithOrgUrl(fmt.Sprintf("https://%s", domain)),
-		okta.WithToken(apiToken),
-		okta.WithHttpClientPtr(client),
-		okta.WithCache(false),
-	)
-	if err != nil {
-		return nil, err
+	if apiToken != "" && domain != "" {
+		_, oktaClient, err = okta.NewClient(ctx,
+			okta.WithOrgUrl(fmt.Sprintf("https://%s", domain)),
+			okta.WithToken(apiToken),
+			okta.WithHttpClientPtr(client),
+			okta.WithCache(false),
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if oktaClientId != "" && oktaPrivateKey != "" && domain != "" {
```

Comment:
Shouldn't this be an else if? Also we should have an else that returns an error if none of these are set.